### PR TITLE
[Refactoring] get rid of some MailboxMessage child class

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Comment by chibenwa** [comment]
`Caused by: com.datastax.driver.core.exceptions.InvalidQueryException: Keyspace 'apache_james' does not exist`

Rebase needed

**Comment by chibenwa** [comment]
Was :green_apple: , rebased to solve simple conflicts...

**Comment by chibenwa** [comment]
Merged

File: mailbox/store/src/test/java/org/apache/james/mailbox/store/MessageBuilder.java [hunk 1]
```diff
@@ -45,7 +59,26 @@ public MailboxMessage build() throws Exception {
...
     public MailboxMessage build(MessageId messageId) throws Exception {
-        return new SimpleMailboxMembership(messageId, mailboxId, uid, -1,  internalDate, size, flags, body, headers);
+        byte[] headerContent = getHeaderContent();
+        SimpleMailboxMessage mailboxMessage = new SimpleMailboxMessage(messageId, internalDate, size, headerContent.length,
+            new SharedByteArrayInputStream(Bytes.concat(headerContent, body)), flags, new PropertyBuilder(), mailboxId, NO_ATTACHMENTS);
+        mailboxMessage.setUid(uid);
+        return mailboxMessage;
+    }
+
+    private byte[] getHeaderContent() throws IOException {
```
Comment (mbaechler) [inline mailbox/store/src/test/java/org/apache/james/mailbox/store/MessageBuilder.java]: This class is not tested AFAIK, the code you wrote (or probably copy/paste) is not proved to work, right?

File: mailbox/lucene/src/test/java/org/apache/james/mailbox/lucene/search/LuceneMailboxMessageSearchIndexTest.java [hunk 2]
```diff
@@ -123,32 +124,60 @@ public void setUp() throws Exception {
         headersTestSubject.put("Cc", "test211 <test21@localhost>, test6 <test6@foobar>");
         
         uid1 = MessageUid.of(1);
-        SimpleMailboxMembership m = new SimpleMailboxMembership(id1, (TestId) mailbox.getMailboxId(), uid1, 0, new Date(), 200, new Flags(Flag.ANSWERED), "My Body".getBytes(), headersSubject);
-        index.add(session, mailbox, m);
+        MessageBuilder builder1 = new MessageBuilder()
```
Comment (mbaechler) [inline mailbox/lucene/src/test/java/org/apache/james/mailbox/lucene/search/LuceneMailboxMessageSearchIndexTest.java]: builder1?

**Pull request closed without merging**